pkg/cmd/experimental/bundlesecret/bundle-secret.go

```diff
@@ -0,0 +1,117 @@
+	"io"
+	"io/ioutil"
+	"os"
+	"strings"
+)
+
+// MaxFileSize?
+const MaxSecretSize = 1 * 1024 * 1024
+const MaxFileSize = 1 * 1024 * 1024
+
+func NewCmdBundleSecret(f *clientcmd.Factory, parentName, name string, out io.Writer) *cobra.Command {
+	var sourcelist util.StringList
+	cmd := &cobra.Command{
+		Use:   "bundle-secret NAME -f SOURCENAME",
+		Short: "bundle data from a source list to load kubernetes secret",
+		Long:  "bundle data from a source list to load kubernetes secret",
+		Run: func(cmd *cobra.Command, args []string) {
+			var err error
+			if len(args) == 0 {
+				err = fmt.Errorf("Must specify NAME as an argument")
+				checkErr(err)
+			}
+			objname := args[0]
+			if len(sourcelist) == 0 {
+				err = fmt.Errorf("Must specify files and/or directories to gather data from.")
+				checkErr(err)
+			}
+			for _, isspace := range sourcelist {
+				if isspace == "" {
+					err = fmt.Errorf("No spaces allowed.  Separate source names with commas, no spaces.")
+					checkErr(err)
+				}
+			}
+			secretdata := make(map[string][]byte)
+			for _, sourceitem := range sourcelist {
+				sourceitem = strings.TrimSuffix(sourceitem, "/")
+				info, err := os.Stat(sourceitem)
+				checkErr(err)
+				if info.IsDir() {
+					filelist, err := ioutil.ReadDir(sourceitem)
+					checkErr(err)
+					for _, item := range filelist {
+						if !item.IsDir() {
+							readValidFiles(sourceitem, item.Name(), item.Size(), secretdata)
+						}
+					}
+				} else {
+					readValidFiles(sourceitem, info.Name(), info.Size(), secretdata)
+				}
+			}
+			secretobj := kapi.Secret{
+				ObjectMeta: kapi.ObjectMeta{Name: objname},
+				Data:       secretdata,
+			}
+			totalsize := 0
+			for _, bytes := range secretdata {
+				totalsize += len(bytes)
+			}
+			if totalsize == 0 {
+				err = fmt.Errorf("No data added to secret.  No files read.")
+				checkErr(err)
+			}
+			if totalsize > MaxSecretSize {
+				err = fmt.Errorf("Secret exceeds max size")
+				checkErr(err)
+			}
+			setDefaultPrinter(cmd)
+			err = f.Factory.PrintObject(cmd, &secretobj, out)
+			checkErr(err)
+		},
+	}
+	cmd.Flags().VarP(&sourcelist, "source", "f", "Comma-separated list of filenames, directories to use as source of secret data")
+	kcmdutil.AddPrinterFlags(cmd)
+	return cmd
+}
+func setDefaultPrinter(c *cobra.Command) {
+	flag := c.Flags().Lookup("output")
+	if len(flag.Value.String()) == 0 {
+		flag.Value.Set("json")
+	}
+}
+func checkErr(err error) {
+	if err == nil {
+		return
+	}
+	fmt.Fprintf(os.Stderr, "Error: %v\n", err)
+	os.Exit(1)
+}
+
+func readValidFiles(fullpath string, filename string, filesize int64, datamap map[string][]byte) {
+	file := fmt.Sprint(fullpath, "/", filename)
+	if !util.IsDNS1123Subdomain(filename) {
+		fmt.Printf("\n%s is not a valid DNS Subdomain, skipping...\n", file)
+		return
+	}
+	if binary.Size(filesize) > MaxFileSize {
+		fmt.Printf("\n%s file size exceeds max, skipping...\n", file)
+		return
+	}
+	data, err := ioutil.ReadFile(file)
+	checkErr(err)
```

Comment:
if you encounter an error, return it, don't do something that causes os.Exit() to happen down inside a command